src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/integration/SessionTemplateSQLGeneratorTest.kt

```diff
@@ -128,6 +130,149 @@ class SessionTemplateSQLGeneratorTest() {
     }
   }
 
+  @Test()
+  fun `Data parsed - session template levelOne validated correctly`() {
+
+    // Given
+    val sessionTemplateSQLGenerator = SessionTemplateSQLGenerator()
+    val sessionLocationList = mutableListOf<SessionLocationColumns>()
+
+    val sessionLocationColumns = mock(SessionLocationColumns::class.java)
+    Mockito.`when`(sessionLocationColumns.levelOne).thenReturn(listOf())
+    Mockito.`when`(sessionLocationColumns.prison).thenReturn("prison1")
+    Mockito.`when`(sessionLocationColumns.key).thenReturn("key1")
+    sessionLocationList.add(sessionLocationColumns)
+
+    // When
+    val exception = assertThrows(IllegalArgumentException::class.java) {
+      sessionTemplateSQLGenerator.validateSessionLocation(sessionLocationList)
+    }
+
+    // Then
+    assertThat(exception.message).isEqualTo("Location : must have at least one level one element (prison:prison1 key:key1)!")
+  }
+
+  @Test()
+  fun `Data parsed - session template levelTwo cant have to parents validated correctly`() {
```

Comment:
sorry very minor comment - i think you mean two.

Reply:
Will correct Sir ;-)